Framework/PythonInterface/plugins/algorithms/BilbySANSDataProcessor.py

```diff
@@ -0,0 +1,603 @@
+            q1d = self._q1d(ws_sam, binning_q, pixeladj, wavelengthadj, wavepixeladj, account_for_gravity,
+                            solid_angle_weighting, radiuscut, wavecut, extralength, qresolution)
+
+            self.setProperty("OutputWorkspace", q1d)  # set output, file 1D pattern
+
+    def _get_time_span(self, ws):
+        run = ws.getRun()
+        duration = run.endTime() - run.starme()
+        return float(duration.total_microseconds())
+
+    def _get_bm_counts(self, ws):
+        return float(ws.run().getProperty("bm_counts").value)
+
+    def _get_frame_count(self, ws):
+        return float(ws.run().getProperty("frame_count").value)
+
+    def _get_period(self, ws):
+        return float(ws.run().getProperty("period").value)
+
+    def _get_l1(self, ws):
+        return float(ws.run().getProperty("L1").value)
+
+    def _apply_mask(self, ws, mask):
+        alg = self.createChildAlgorithm("MaskDetectors")
+        alg.setProperty("Workspace", ws)
+        alg.setProperty("MaskedWorkspace", mask)
+        alg.execute()
+
+    def _convert_units(self, ws, unit):
+        alg = self.createChildAlgorithm("ConvertUnits")
+        alg.setProperty("InputWorkspace", ws)
+        alg.setProperty("Target", unit)
+        alg.execute()
+
+        return alg.getProperty("OutputWorkspace").value
+
+    def _rebin(self, ws, binning, preserveevents):
+        alg = self.createChildAlgorithm("Rebin")
+        alg.setProperty("InputWorkspace", ws)
+        alg.setProperty("Params", binning)
+        alg.setProperty("PreserveEvents", preserveevents)
+        alg.execute()
+
+        return alg.getProperty("OutputWorkspace").value
+
+    def _multiply(self, a, b):
+        alg = self.createChildAlgorithm("Multiply")
+        alg.setProperty("LHSWorkspace", a)
+        alg.setProperty("RHSWorkspace", b)
+        alg.execute()
+
+        return alg.getProperty("OutputWorkspace").value
+
+    def _scale_mult(self, ws_input, factor, operation):
+        alg = self.createChildAlgorithm("Scale")
+        alg.setProperty("InputWorkspace", ws_input)
+        alg.setProperty("Factor", factor)
+        alg.setProperty("Operation", operation)
+        alg.execute()
+        return alg.getProperty("OutputWorkspace").value
+
+    def _subtract(self, a, b):
+        alg = self.createChildAlgorithm("Minus")
+        alg.setProperty("LHSWorkspace", a)
+        alg.setProperty("RHSWorkspace", b)
+        alg.execute()
+        return alg.getProperty("OutputWorkspace").value
+
+    def _single_valued_ws(self, value):
+        alg = self.createChildAlgorithm("CreateSingleValuedWorkspace")
+        alg.setProperty("DataValue", value)
+        alg.execute()
+        return alg.getProperty("OutputWorkspace").value
+
+    def _mask_to_roi(self, ws_mask):
+        # invert mask and then extract "masked" detectors in order to get ROI
+        # BUG in Mantid forces us to use AnalysisDataService
+        alg = AlgorithmManager.create("InvertMask")
```

Comment:
Could I ask why this call isn't being made just on one line (e.g. InvertMask(InputWorkspaces=ws_mask, OutputWorkspace="_ws") instead of using the AlgorithmManager? (Mostly for my own blind curiosity)

Reply:
There is not as far as I am aware any reason to not do it on one line.

Reply:
I just did not know how to make it differently.